src/main/java/seedu/address/model/module/TutorialTeam.java

```diff
@@ -0,0 +1,190 @@
+
+import seedu.address.model.person.Person;
+
+/**
+ * Represents a Module's tutorial team.
+ * Guarantees: immutable; is valid as declared in
+ * {@link #isValidTeamName(String)}
+ */
+public class TutorialTeam {
+
+    public static final String MESSAGE_CONSTRAINTS = "Please enter a valid tutorial team name "
+            + "eg. Team 1, and it should not be blank";
+    public static final String MESSAGE_NAME_CONSTRAINTS = "Team names should only contain alphanumeric "
+            + "characters and spaces, and it should not be blank";
+    public static final String MESSAGE_SIZE_CONSTRAINTS = "Team size should be a positive integer";
+
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    public final String teamName;
+    public final int teamSize;
+    private final ArrayList<Person> students;
+
+    /**
+     * Constructs a {@code TutorialTeam} with default values.
+     * Initializes the {@code value} field to an empty string, creates an empty
+     * list for {@code students} and sets the {@code teamSize} to the maximum.
+     */
+    public TutorialTeam() {
+        this.teamName = "";
+        this.students = new ArrayList<>();
+        this.teamSize = Integer.MAX_VALUE;
+    }
+
+    /**
+     * A constructor for TutorialTeam. Creates a tutorial team of a certain size
+     * with no students.
+     * @param tutorialTeam
+     * @param teamSize
+     */
+    public TutorialTeam(String tutorialTeam, int teamSize) {
+        requireAllNonNull(tutorialTeam, teamSize);
+        checkArgument(isValidTeamName(tutorialTeam), MESSAGE_NAME_CONSTRAINTS);
+        checkArgument(isValidSize(teamSize), MESSAGE_SIZE_CONSTRAINTS);
+        this.teamName = tutorialTeam;
+        this.students = new ArrayList<>();
+        this.teamSize = teamSize;
+    }
+
+    /**
+     * A constructor for TutorialTeam. Creates a tutorial team with students.
+     * @param tutorialClass to be added
+     * @param students      in the tutorial class
+     */
+    public TutorialTeam(String tutorialTeam, ArrayList<Person> students) {
+        requireAllNonNull(tutorialTeam);
+        checkArgument(isValidTeamName(tutorialTeam), MESSAGE_NAME_CONSTRAINTS);
+        this.teamName = tutorialTeam;
+        this.students = students;
+        this.teamSize = Integer.MAX_VALUE;
+    }
+
+    /**
+     * A constructor for TutorialTeam. Creates a tutorial team with students and
+     * team size.
+     * @param tutorialClass to be added
+     * @param students      in the tutorial class
+     * @param teamSize      of the tutorial team
+     */
+    public TutorialTeam(String tutorialTeam, ArrayList<Person> students, int teamSize) {
+        requireAllNonNull(tutorialTeam);
+        checkArgument(isValidTeamName(tutorialTeam), MESSAGE_NAME_CONSTRAINTS);
+        checkArgument(isValidSize(teamSize), MESSAGE_SIZE_CONSTRAINTS);
+        this.teamName = tutorialTeam;
+        this.students = students;
+        this.teamSize = teamSize;
```

Comment:
Owing to the comment above for this class, should we put a limit on the team size?

Reply:
This can be up for discussion. Be in general, I dont think we need to put a limit on team size since we cannot practically determine an appropriate one-size-fits-all team size.

Reply:
Also, having a limit on team size generally means we need to impose a size restriction on a tutorial class too since it is a subset of a tutorial class